source/contribute.rst

```diff
@@ -41,128 +41,150 @@ contributing back to a Doctrine project.
 Initial Setup
 -------------
 
+-  Clone the project
 -  Setup a `GitHub <https://github.com>`_ account.
 -  Fork the repository of the project you want to contribute to. In this example
    it will be `DBAL <https://github.com/doctrine/dbal>`_
--  Clone your fork locally
+-  Make the local repository aware of your fork.
 
 .. code-block:: console
 
-    $ git clone git@github.com:username/dbal.git
+    $ git clone --origin doctrine git@github.com:doctrine/dbal.git
 
--  Enter the dbal directory and add the **doctrine** remote
+At that point, you have a repository, and you should be able to start
+doing changes. Later on, if you end up with something you are satisfied
+with, you will need to be able to push your changes to your fork.
+
+Here is how you can make your local repository aware of your fork:
 
 .. code-block:: console
 
     $ cd dbal
-    $ git remote add doctrine git://github.com/doctrine/dbal.git
+    $ git remote add fork git@github.com:username/dbal.git
 
-Branching from the default branch
----------------------------------
+You should now have two remotes: one called ``doctrine``, another called
+``fork``.
 
-New pull requests are created with the repository's default branch as base branch.
-The default branch is the branch you see when you enter the repository page on GitHub.
+Choosing the right branch
+-------------------------
 
-.. image:: ../images/default-branch.png
-   :alt: The default branch
-   :style: margin-bottom: 20px
+Before you start working on a feature or a bug fix, you need to choose
+which branch you are going to contribute to. We try hard to follow
+semver, which means there are three choices.
 
-In this DBAL example, it's the branch with the name **2.11.x**. The branch name reflects the
-current lowest supported version of a repository.
+Let's say the latest version of the project is 2.3.1. The following
+branches should exist and are the possible choices:
```

Comment:
What would be the best way to find out for a given Doctrine project what the current stable/bugfix/... version is?

Along the lines of...
--
You can find out at https://www.doctrine-project.org/projects/orm.html [<-- applies to ORM only] what the current stable as well as the upcoming minor and major versions are.

For example, let's say the current stable version is 2.3.1. This makes 2.4 as well as 3.0 the next minor and major version releases. In the repository, there will be corresponding branches:

Reply:
We sometimes forget to update the project metadata, so the best way is really to find what the latest (or rather, highest) version number is, and derive the branch information from that.

Reply:
There isn't AFAIK. I should create another PR to https://www.doctrine-project.org/contribute/maintainer/index.html I suppose.

Reply:
✅